### PR TITLE
Add entity types api doc

### PR DESCRIPTION
**Pull request by mooreds**
This is the docs for the entity type API.

### CODE REVIEW

File: site/docs/src/json/entity-types/response.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    "lastUpdateInstant": 1595361143101,
+    "name": "Customer",
+    "permissions": [
+      {
+        "entityTypeId": "8174f72f-5ecd-4eae-8de8-7fef597b3473",
+        "id": "ce485a91-906f-4615-af75-81d37dc71e90",
+        "insertInstant": 1595361142909,
+        "isDefault": false,
+        "lastUpdateInstant": 1595361143101,
+        "name": "admin"
```
Comment (bguyza) [inline site/docs/src/json/entity-types/response.json:22]: `entityTypeId` will not be in the permission json.  Here is an example with all fields:

```
{
  "id": "ce485a91-906f-4615-af75-81d37dc71e90",
  "name": "admin",
  "description": "This permission grants admin privileges",
  "isDefault": false,
  "insertInstant": 1595361142909,      
  "lastUpdateInstant": 1595361143101
}
```

File: site/docs/src/json/entity-types/response.json [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      },
+      {
+        "id": "ce485a91-906f-4615-af75-81d37dc71e91",
+        "name": "user",
+        "isDefault": true
+      }
```
Comment (bguyza) [inline site/docs/src/json/entity-types/response.json:28]: `insertInstant` and `lastUpdateInstant` are required so I do not think we should omit them here.

File: site/docs/src/json/entity-types/responses.json [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+          "id": "ce485a91-906f-4615-af75-81d37dc71e90",
+          "insertInstant": 1595361142909,
+          "isDefault": false,
+          "lastUpdateInstant": 1595361143101,
+          "name": "admin"
+        },
+        {
+          "id": "ce485a91-906f-4615-af75-81d37dc71e91",
+          "name": "user",
+          "isDefault": true
```
Comment (bguyza) [inline site/docs/src/json/entity-types/responses.json:28]: Same comments from `response.json` applies here.

File: site/docs/v1/tech/apis/_entity-type-request-body.adoc [hunk 4]
```diff
@@ -0,0 +1,27 @@
+==== Request Body
+
+[.api]
+[field]#entityType.data# [type]#[Object]# [optional]#Optional#::
+An object that can hold any information about the Entity Type that should be persisted.
+
+[field]#entityType.jwtConfiguration.accessTokenKeyId# [type]#[UUID]# [optional]#Optional#::
+The unique id of the signing key used to sign the access token.
++
+Required when `enabled` is set to `true`.
```
Comment (bguyza) [inline site/docs/v1/tech/apis/_entity-type-request-body.adoc:10]:
```suggestion
Required when [field]#enabled# is set to `true`.
```

File: site/docs/v1/tech/apis/_entity-type-request-body.adoc [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+[field]#entityType.jwtConfiguration.enabled# [type]#[Boolean]# [optional]#Optional# [default]#defaults to `false`#::
+Indicates if this application is using the JWT configuration defined here or the global JWT configuration defined by the Tenant. If this is `false` the signing algorithm configured in the Tenant will be used. If `true` the signing algorithm defined in this application will be used. 
+
+[field]#entityType.jwtConfiguration.timeToLiveInSeconds# [type]#[Integer]# [optional]#Optional#::
+The length of time in seconds the JWT will live before it is expired and no longer valid.
++
+Required when `enabled` is set to `true`.
```
Comment (bguyza) [inline site/docs/v1/tech/apis/_entity-type-request-body.adoc:18]:
```suggestion
Required when [field]#enabled# is set to `true`.
```

File: site/docs/v1/tech/apis/_entity-type-response-body-base.adoc [hunk 6]
```diff
@@ -0,0 +1,63 @@
+[api]
+ifeval::["{base_field_name}" != "entityType"]
+[field]#{base_field_name}# [type]#[Array]#::
+The list of EntityType objects.
+endif::[]
+
+[field]#{base_field_name}.data# [type]#[Object]# [optional]#Optional#::
```
Comment (bguyza) [inline site/docs/v1/tech/apis/_entity-type-response-body-base.adoc:7]: I believe we omit Optional, Required, and "defaults to" in response body documentation.

File: site/docs/src/json/entity-types/response.json (hunk 1 above)
Comment (mooreds) [reply site/docs/src/json/entity-types/response.json:22]: Hiya @bguyza 

Ah, my bad: https://github.com/FusionAuth/fusionauth-java-client/blob/master/src/main/java/io/fusionauth/domain/EntityTypePermission.java looks like it lists everything.

File: site/docs/v1/tech/apis/_entity-type-response-body-base.adoc (hunk 6 above)
Comment (mooreds) [reply site/docs/v1/tech/apis/_entity-type-response-body-base.adoc:7]: 100%, thanks!

File: site/docs/src/json/entity-types/permission-request.json [hunk 7]
```diff
@@ -0,0 +1,6 @@
+{
+  "permission": {
+    "name": "read",
+    "isDefault": true
```
Comment (bguyza) [inline site/docs/src/json/entity-types/permission-request.json:4]:
```suggestion
    "isDefault": true,
    "description": "The permission description"
```

File: site/docs/src/json/entity-types/permission-response.json [hunk 8]
```diff
@@ -0,0 +1,9 @@
+{
+  "permission": {
+    "id": "ce485a91-906f-4615-af75-81d37dc71e90",
+    "insertInstant": 1595361142909,
+    "lastUpdateInstant": 1595361143101,
+    "name": "read",
```
Comment (bguyza) [inline site/docs/src/json/entity-types/permission-response.json:6]:
```suggestion
    "name": "read",
    "description": "The permission description"
```

File: site/docs/v1/tech/apis/entity-types.adoc [hunk 9]
```diff
@@ -0,0 +1,376 @@
...
+The Id of the Entity Type.
+
+[field]#permissionId# [type]#[UUID]# [optional]#Optional# [default]#defaults to secure random UUID#::
+The Id to use for the new permission. If not specified a secure random UUID will be generated.
+
+==== Request Body
+
+[.api]
+[field]#permission.name# [type]#[String]# [required]#Required#::
+The name of the Permission.
```
Comment (bguyza) [inline site/docs/v1/tech/apis/entity-types.adoc:258]:
```suggestion
The name of the Permission.

[field]#permission.description# [type]#[String]# [optional]#Optional#::
The description of the Permission.

```

File: site/docs/v1/tech/apis/_permission-response-body.adoc [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+The Id of the Permission.
+
+[field]#permission.insertInstant# [type]#[Long]#::
+The link:/docs/v1/tech/reference/data-types/#instants[instant] that the Permission was added to the FusionAuth database.
+
+[field]#permission.lastUpdateInstant# [type]#[Long]#::
+The link:/docs/v1/tech/reference/data-types/#instants[instant] that the Permission was updated in the FusionAuth database.
+
+[field]#permission.name# [type]#[String]#::
+The name of the Permission.
```
Comment (bguyza) [inline site/docs/v1/tech/apis/_permission-response-body.adoc:14]:
```suggestion
The name of the Permission.

[field]#permission.description# [type]#[String]#::
The description of the Permission.
```

**Review by bguyza — COMMENTED** [review]
Looks good.  Just a few additional things:
1. I added suggestions to add "description" to the permission API examples
2. I do believe the fields listed in the requests/responses should be in alphabetical order.  There are a few spots to reorganize.

**Pull request merged**